docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,488 @@
+
+##### Attester generates the evidence (inside an enclave/TEE)
+```C
+...
+result = oe_get_evidence(
+    OE_EVIDENCE_FLAGS_REMOTE_ATTESTATION,
+    NULL, // custom_claims
+    0,
+    NULL, // opt_params
+    0,
+    &evidence,
+    &evidence_size,
+    &endorsements,
+    &endorsements_size);
+...
+```
+
+##### Verifier verifies the evidence with custom endorsements (in the untrusted host or inside an enclave/TEE)
+```C
+...
+//
+// Verifier gets endorsements not using OE SDK
+//
+
+//
+// Verifier builds **endorsements** structure
+//
+endorsements_external = ...
+endorsements_external_size = ...
+
+// Verify evidence with external endorsements
+result = oe_verify_evidence(
+            report,
+            report_size,
+            endorsements_external,
+            endorsements_external_size,
+            NULL,  // opt_params
+            0,
+            &claims,
+            &claims_size);
+...
+```
+
+Specification
+-------------
+
+### Public type definitions
+Generic serializable public structure that stores the endorsements in raw binary format.
+
+`attestation.h`
+```C
+/**
+ * Flags passed to oe_get_evidence() function.
+ */
+#define OE_EVIDENCE_FLAGS_LOCAL_ATTESTATION     0x00000000
+#define OE_EVIDENCE_FLAGS_REMOTE_ATTESTATION    0x00000001
+
+/*! Limit the size of the endorsements */
+#define OE_ATTESTATION_ENDORSEMENT_MAX_SIZE     (20 * 1024)
+
+/*! Endorsement structure version */
+#define OE_ATTESTATION_ENDORSEMENT_VERSION      (1)
+
+/*! \struct oe_endorsements_t
+ *
+ * \brief OE endorsements
+ *
+ * Raw generic serializable structure that contains the endorsements. All
+ * data should be in little endian format.
+ *
+ */
+typedef struct _oe_endorsements_t
+{
+    uint32_t version;       ///< Version of this structure
+    uint32_t enclave_type;  ///< The type of enclave (oe_enclave_type_t)
+    uint32_t buffer_size;   ///< Size of the buffer
+    uint32_t num_elements;  ///< Number of elements stored in the data buffer
+
+    /*! Data buffer is made of an offset array of type uint32_t, followed by
+     * the actual data.
+     * This array has the size of **num_elements** and stores the offset
+     * into the data section.
+     * _________________________
+     * |  version              |
+     * |-----------------------|
+     * |  enclave_type         |
+     * |-----------------------|
+     * |  buffer_size          |
+     * |-----------------------|
+     * |  num_elements         |
+     * |-----------------------|
+     * |  offsets              |
+     * |  (array of uint32_t   |
+     * |  with length of       |
+     * |  num_elements)        |
+     * |-----------------------|
+     * |  buffer (data)        |
+     * |_______________________|
+     */
+    uint8_t buffer[];              ///< Buffer of offsets + data
+
+} oe_endorsements_t;
+
+/*! Version of the supported SGX endorsement structures */
+#define OE_SGX_ENDORSEMENTS_VERSION     (1)
```

Comment:
Consider moving SGX specific definitions outside of the TEE agnostic header.

Reply:
i moved them